packages/core/src/useNavigationCache.tsx

```diff
@@ -133,6 +140,13 @@ export default function useNavigationCache<
         // FIXME: too much work to fix the types for now
         ...(emitter.create(route.key) as any),
         dispatch: (thunk: Thunk) => withStack(() => dispatch(thunk)),
+        getParent: (id?: string) => {
+          if (id !== undefined && id === rest.getId()) {
+            return acc[route.key];
+          }
+
+          return rest.getParent(id);
+        },
```

Comment:
just want to verify that this logic is correct - might be worth adding a comment to clarify intent, or maybe it's obvious to someone who is more familiar with the codebase these days ;)